### ISSUE TITLE
overfitting

### ISSUE THREAD

**Issue by xuchengggg** [dbolya/yolact#148]
Hi, I implemented your model in tf.keras. But I have encountered some overfitting problem. I trained on COCO train 2017 and use same data augmentation as you, and set the L2  norm alpha 0.0005. And there is a problem, in keras, L2 loss is added in total loss, which means `total_loss=L2_loss + loc_loss + class_loss + mask_loss` (I didn't use seg_loss). I have no ider about pytorch, since in your loss, `total_loss = loc_loss + class_loss + mask_loss + seg_loss`.  So I don't know if your loss contains L2 loss.
In my training process, when about in 160K steps, the mAP in val 2017 can reach about 17, but after that, val loss will go up, but the train loss still go down.  Could you give me some suggestions?

**Comment by dbolya** [comment]
In Pytorch, regularization is called "weight decay", though it might actually be L1 loss idk. The alpha I have set for that is `5e-4` which is exactly what you're using.

As for the mAP after 160K iterations, that should be *much* higher than 17. Actually, depending on the random initialization I expect the model to achieve 15-16 mAP after just the second epoch. Iteration 160K is around the 10th epoch, which should be 19-20 mAP, so it seems like you might have some implementation errors? Are you training from imagenet pretrained weights? All these mAPs were for the resnet101 version btw.
